src/Compilers/CSharp/Portable/Compilation/SyntaxAndDeclarationManager.cs

```diff
@@ -150,19 +160,22 @@ public SyntaxAndDeclarationManager AddSyntaxTrees(IEnumerable<SyntaxTree> trees)
             IDictionary<SyntaxTree, ImmutableArray<LoadDirective>> loadDirectiveMapBuilder,
             IDictionary<string, SyntaxTree> loadedSyntaxTreeMapBuilder,
             IDictionary<SyntaxTree, Lazy<RootSingleNamespaceDeclaration>> declMapBuilder,
+            IDictionary<SyntaxTree, OneOrMany<ImmutableSegmentedHashSet<string>>> lastComputedMemberNamesMap,
             ref DeclarationTable declTable)
         {
             var sourceCodeKind = tree.Options.Kind;
             if (sourceCodeKind == SourceCodeKind.Script)
             {
-                AppendAllLoadedSyntaxTrees(treesBuilder, tree, scriptClassName, resolver, messageProvider, isSubmission, ordinalMapBuilder, loadDirectiveMapBuilder, loadedSyntaxTreeMapBuilder, declMapBuilder, ref declTable);
+                AppendAllLoadedSyntaxTrees(treesBuilder, tree, scriptClassName, resolver, messageProvider, isSubmission, ordinalMapBuilder, loadDirectiveMapBuilder, loadedSyntaxTreeMapBuilder, declMapBuilder, lastComputedMemberNamesMap, ref declTable);
             }
 
-            AddSyntaxTreeToDeclarationMapAndTable(tree, scriptClassName, isSubmission, declMapBuilder, ref declTable);
+            AddSyntaxTreeToDeclarationMapAndTable(
+                tree, scriptClassName, isSubmission, declMapBuilder,
+                lastComputedMemberNames: OneOrMany<ImmutableSegmentedHashSet<string>>.Empty, ref declTable);
```

Comment:
>OneOrMany<ImmutableSegmentedHashSet<string>>.Empty [](http://example.com/codeflow?start=41&length=50)

It is not obvious why this is the right value.

Reply:
It's a place where we do not apply the optimization.  Right now, we do it for editing scenarios where we can easily know the prior and tree we're replacing.  Other compilation manipulation scenarios weren't a concern. 

I can doc this and mention this could be changed in the future if there was a desired scenario driving it.

Reply:
doc'ed